ibis/backends/pyspark/tests/conftest.py

```diff
@@ -177,48 +204,139 @@ def connect(*, tmpdir, worker_id, **kw):
         return ibis.pyspark.connect(spark, **kw)
 
 
-@pytest.fixture(scope="session")
-def con(data_dir, tmp_path_factory, worker_id):
-    import pyspark.sql.functions as F
+class TestConfForStreaming(TestConf):
+    # Note: The same session can be used for both batch and streaming
+    # jobs in Spark. Streaming is made explicit on the source
+    # dataframes. This is why, we do not really need a separate
+    # `TestConf` class for streaming, but only need to create
+    # streaming counterparts for the test tables. Still added this
+    # class to keep the testing uniform with Flink. This class is used
+    # in `con_streaming()` fixture (streaming counterpart of `con()`)
+    # to create a new spark session and load the `***_streaming`
+    # tables for testing.  However, either `con()` or
+    # `con_streaming()` can be used to execute any batch/streaming
+    # job. This is why, we set `autouse=True` for `con_streaming()`
+    # to create the streaming tables, and then rely solely on `con()`
+    # to operate on those tables in the tests.
```

Comment:
It makes sense to try to keep the testing uniform, but I also wonder if it's cleaner/easier to just reuse the same test con, given this difference in behavior... It feels a little unncessary. Maybe others have some thoughts.